feat: constraint weight override (#946)
TimefoldAI/timefold-solver: docs/src/modules/ROOT/pages/constraints-and-score/constraint-configuration.adoc

----
@PlanningSolution
public class VehicleRoutePlan {

    ...

    ConstraintWeightOverrides<HardSoftScore> constraintWeightOverrides;

    void setConstraintWeightOverrides(ConstraintWeightOverrides<HardSoftScore> constraintWeightOverrides) {
        this.constraintWeightOverrides = constraintWeightOverrides;
    }

    ConstraintWeightOverrides<HardSoftScore> getConstraintWeightOverrides() {
        return constraintWeightOverrides;
    }

    ...
}
----

We've just introduced a new field of type `ConstraintWeightOverrides`,
and we provided a getter and a setter for it.
The field will be automatically exposed as a xref:using-timefold-solver/modeling-planning-problems.adoc#problemFacts[problem fact],
there is no need to add a `@ProblemFactProperty` annotation.
But we need to fill it with the desired constraint weights:

[source,java,options="nowrap"]
----
...

var constraintWeightOverrides = ConstraintWeightOverrides.of(
    Map.of(
        "Vehicle capacity", HardSoftScore.ofHard(2),
        "Service finished after max end time", HardSoftScore.ZERO
    )
);

var solution = new VehicleRoutePlan();
solution.setConstraintWeightOverrides(constraintWeightOverrides);

...
----

> The `Vehicle capacity` constraint in this planning solution will get a weight of `2hard`,

Reviewer: ```suggestion
The `Vehicle capacity` constraint in this planning solution has a weight of `2hard`,
```

as opposed to its original `1hard`.

> The `Service finished after max end time` constraint will have a weight of `0hard`,

Reviewer: ```suggestion
The `Service finished after max end time` constraint has a weight of `0hard`,
```

and therefore will be disabled entirely.

> This way, you can solve the same problem
> while applying different constraint weights to each instance.

Reviewer: ```suggestion
In this way, you can solve the same problem
by applying different constraint weights to each instance.
```
